packages/core/src/model.ts

```diff
@@ -52,20 +52,32 @@ export class ChartModel {
 		const { groupMapsTo } = this.getOptions().data;
 
 		const axesOptions = this.getOptions().axes;
-		
+
 		// Check for custom domain
 		if (axesOptions) {
-			Object.keys(axesOptions).forEach(axis => {
-				if (axesOptions[axis].domain && axesOptions[axis].mapsTo) {
+			Object.keys(axesOptions).forEach((axis) => {
+				if (
+					axesOptions[axis].mapsTo &&
+					axesOptions[axis].domain &&
+					!this.getOptions().percentage
+				) {
 					const mapsTo = axesOptions[axis].mapsTo;
-					const [start, end] = axesOptions[axis].domain;
 
-					// Filter out data outside domain
-					displayData = displayData.filter((datum) => 
-						datum[mapsTo] >= start && datum[mapsTo] <= end
-					)
+					if (axesOptions[axis].scaleType === ScaleTypes.LABELS) {
+						displayData = displayData.filter((datum) =>
+							axesOptions[axis].domain.includes(datum[mapsTo])
+						);
+					} else {
+						const [start, end] = axesOptions[axis].domain;
+
+						// Filter out data outside domain
+						displayData = displayData.filter(
+							(datum) =>
+								datum[mapsTo] >= start && datum[mapsTo] <= end
```

Comment:
```suggestion
						displayData = displayData.filter((datum) =>
							datum[mapsTo] >= start && datum[mapsTo] <= end
```

Reply:
It's prettier formatting.

Reply:
I don't get it why wouldn't it do it for the filter above??